pkg/kubelet/util/util_windows_test.go

```diff
@@ -31,17 +32,83 @@ import (
 	"github.com/stretchr/testify/require"
 )
 
+func TestGetAddressAndDialer(t *testing.T) {
+
+	// Compare dialer function by pointer
+	tcpDialPointer := reflect.ValueOf(tcpDial).Pointer()
+	npipeDialPointer := reflect.ValueOf(npipeDial).Pointer()
+	var nilDialPointer uintptr = 0x0
+
+	tests := []struct {
+		endpoint      string
+		expectedAddr  string
+		expectedDial  uintptr
+		expectedError bool
+	}{
+		{
+			endpoint:      "tcp://localhost:15880",
+			expectedAddr:  "localhost:15880",
+			expectedDial:  tcpDialPointer,
+			expectedError: false,
+		},
+		{
+			endpoint:      "npipe://./pipe/mypipe",
+			expectedAddr:  "//./pipe/mypipe",
+			expectedDial:  npipeDialPointer,
+			expectedError: false,
+		},
+		{
+			endpoint:      "npipe:\\\\.\\pipe\\mypipe",
+			expectedAddr:  "//./pipe/mypipe",
+			expectedDial:  npipeDialPointer,
+			expectedError: false,
+		},
+		{
+			endpoint:      "unix:///tmp/s1.sock",
+			expectedAddr:  "",
+			expectedDial:  nilDialPointer,
+			expectedError: true,
+		},
+		{
+			endpoint:      "tcp1://abc",
+			expectedAddr:  "",
+			expectedDial:  nilDialPointer,
+			expectedError: true,
+		},
+		{
+			endpoint:      "a b c",
+			expectedAddr:  "",
+			expectedDial:  nilDialPointer,
+			expectedError: true,
+		},
+	}
+
+	for _, test := range tests {
+		address, dialer, err := GetAddressAndDialer(test.endpoint)
+
+		dialerPointer := reflect.ValueOf(dialer).Pointer()
+		assert.Equal(t, test.expectedDial, dialerPointer)
```

Comment:
When comparing pointers, and you have to do that when comparing funcs), it is always nice with a verbose output that prints something else than the non-matching pointers. What about printing the name of the function?

Reply:
(I find debugging a "this ptr is not the same as this ptr" problem is quite hard.)

Reply:
Good idea!

Reply:
(it always has been)

Reply:
I print it like this:

```
Messages:       Expected dialer nilDial, but get k8s.io/kubernetes/pkg/kubelet/util.npipeDial
Messages:       Expected dialer k8s.io/kubernetes/pkg/kubelet/util.tcpDial, but get k8s.io/kubernetes/pkg/kubelet/util.npipeDial

```